### PR TITLE
Add missing tags to Discovery Page

### PR DESCRIPTION
**Pull request by tiffperumpail**
Fix bug so that nutrition types and Real Food Challenge tags create carousels.

### CODE REVIEW

File: app/models/product.rb [hunk 1]
```diff
@@ -35,6 +35,30 @@ def self.get_tags_hash
...
       tags_hash.update(tag_hash)
     end
+    self.get_dietary_and_rfc_tags(tags_hash)
+  end
+  
+  def self.get_dietary_and_rfc_tags(tags_hash)
+    tags = [["vegan", "Vegan"], ["gluten_free", "Gluten Free"], ["dairy_free", "Dairy Free"], ["lc_based", "Locally Based"], ["fair", "Fairly Traded"], ["eco_sound", "Ecologically Sound"], ["humane", "Humane"]]
+    tags.each do |tag_type, tag_name|
+      tags_hash[tag_name] = []
+    end
```
Comment (ErikOstien) [inline app/models/product.rb:45]: Adding tags to tag hash looks good

File: app/models/product.rb [hunk 2]
```diff
@@ -35,6 +35,30 @@ def self.get_tags_hash
...
+    
+    Product.all.each do |product|
+      tags.each do |tag_type, tag_name|
+        if product[tag_type]
+          tags_hash[tag_name] += [product]
+        end
+      end
+    end
```
Comment (ErikOstien) [inline app/models/product.rb:53]: Add products for each tag looks good

File: app/models/product.rb [hunk 3]
```diff
@@ -35,6 +35,30 @@ def self.get_tags_hash
...
+    
+    tags.each do |tag_type, tag_name|
+      if tags_hash[tag_name].length == 0
+        tags_hash.delete(tag_name)
+      end
+    end
```
Comment (ErikOstien) [inline app/models/product.rb:59]: removing unused tags looks good

File: features/step_definitions/carousel_steps.rb [hunk 4]
```diff
@@ -27,6 +27,15 @@
...
 
+Given /I create four new products.*/ do
+  step %{Given a vendor already exists}
+  FactoryBot.create(:product, name: "First Product")
+  FactoryBot.create(:product, name: "Second Product")
+  FactoryBot.create(:product, name: "Third Product")
+  FactoryBot.create(:product, name: "Four Product")
+end
+
+
```
Comment (Juggernaut265) [inline features/step_definitions/carousel_steps.rb]: The above step def is not needed. There is a step def in index_steps.rb that does this for a dynamic number of vendors/products. "Given/there are (.*)(.*)/" will allow us to create any number of products or vendors. We still need the "step %{Given a vendor already exists}" though.

File: features/carousel.feature [hunk 5]
```diff
@@ -28,4 +28,21 @@ Feature: View and Interact with a Carousel of Products and Vendors
...
+  Scenario: Carousel for products with Real Food Challenge tag
+    Given I create four new products
+    And I am on the Discovery page
+    And I should see a carousel for the type "Humane"
+    Then I should see all the "Humane" products
+
+  Scenario: Don't see carousel for unused nutrition tag
+    Given I create four new products that are not Gluten Free
+    And I am on the Discovery page
+    Then I should not see a carousel for the type "Gluten Free"
```
Comment (ErikOstien) [inline features/carousel.feature:51]: New tests look good and pass in checks

**Review by ErikOstien — APPROVED** [review]
Adding all the missing tags to the carousel looks good and the tests that accompany this addition also look good. Great Job!

File: features/step_definitions/carousel_steps.rb [hunk 6]
```diff
@@ -27,6 +27,15 @@
...
 end
 
+Given /I create four new products.*/ do
+  step %{Given a vendor already exists}
+  FactoryBot.create(:product, name: "First Product")
+  FactoryBot.create(:product, name: "Second Product")
+  FactoryBot.create(:product, name: "Third Product")
+  FactoryBot.create(:product, name: "Four Product")
+end
+
```
Comment (Juggernaut265) [inline features/step_definitions/carousel_steps.rb]: The above step def is not needed. There is a step def in index_steps.rb that does this for a dynamic number of vendors/products. "Given/there are (.*)(.*)/" will allow us to create any number of products or vendors. We still need the "step %{Given a vendor already exists}" though.

**Comment by Juggernaut265** [comment]
Added a question/comment on "features/step_definitions/carousel_steps.rb" in the "Files Changed"

**Review by Juggernaut265 — CHANGES_REQUESTED** [review]
See below

File: features/step_definitions/carousel_steps.rb (hunk 6 above)
Comment (tiffperumpail) [reply features/step_definitions/carousel_steps.rb]: Fixed!

**Review by Juggernaut265 — APPROVED** [review]

**Pull request merged**